pytest_html/plugin.py

```diff
@@ -95,23 +98,27 @@ def __init__(self, logfile):
         self.passed = self.skipped = 0
         self.xfailed = self.xpassed = 0
         self.rerun = 0
+        self.self_contained = self_contained
 
     class TestResult:
 
-        def __init__(self, outcome, report):
+        def __init__(self, outcome, report, self_contained,
+                     logfile, test_index):
```

Comment:
I still don't think we need `test_index` as we already have the report. Can you try using `getattr(report, 'reruns', 0)` in the call to `append_extra_html`?